core/templates/services/messenger.service.ts

```diff
@@ -19,147 +19,196 @@
  * be attempted due to cross-domain security issues.)
  */
 
-angular.module('oppia').factory('MessengerService', [
-  '$log', '$window', function($log, $window) {
-    var isPositiveInteger = function(n) {
-      return (typeof n === 'number' && n % 1 === 0 && n > 0);
-    };
-    var isBoolean = function(b) {
-      return typeof b === 'boolean';
-    };
-
-    var SUPPORTED_HASHDICT_VERSIONS = [
-      '0.0.0', '0.0.1', '0.0.2', '0.0.3'
-    ];
-
-    var MESSAGE_VALIDATORS = {
-      heightChange: function(payload) {
-        return isPositiveInteger(payload.height) && isBoolean(payload.scroll);
-      },
-      explorationLoaded: function() {
-        return true;
-      },
-      stateTransition: function(payload) {
-        return Boolean(payload.oldStateName) || Boolean(payload.newStateName);
-      },
-      explorationReset: function(payload) {
-        return Boolean(payload.stateName);
-      },
-      explorationCompleted: function() {
-        return true;
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+
+import { LoggerService } from 'services/contextual/logger.service';
+import { WindowRef } from 'services/contextual/window-ref.service';
+
+interface IHeightChangeData {
+  height: number;
+  scroll: number;
+}
+
+interface IExplorationLoadedData {
+  explorationVersion: number;
+  explorationTitle: string;
+}
+
+interface IStateTransitionData {
+  explorationVersion: number;
+  oldStateName: string;
+  jsonAnswer: string;
+  newStateName: string;
+}
+
+interface IExplorationCompletedData {
+  explorationVersion: number;
+}
+
+interface IMessageValidatorsType {
+  heightChange(payload: IHeightChangeData): boolean;
+  explorationLoaded(): boolean;
+  stateTransition(payload: IStateTransitionData): boolean;
+  explorationReset(payload: {stateName: string}): boolean;
+  explorationCompleted(): boolean;
+}
+
+interface GetPayloadType {
```

Comment:
This interface hasn't been prefixed with "I". Any special reason. And I couldn't see anything about naming interfaces in the code style guide. Has it been decided that we are going with the C# style of naming interfaces by prefixing them with an "I"? + @bansalnitish

Reply:
Missed adding the prefix. Updated.

Also, I think we should have a convention and a test which ensures that it is being followed.

Reply:
> Also, I think we should have a convention and a test which ensures that it is being followed.

Yeah, I agree. Should we open an issue or just tag someone so that this doesn't get submerged.

Reply:
https://github.com/oppia/oppia/issues/9348

Reply:
Thanks for filing this issue @ankita240796